pkg/receive/handler.go

```diff
@@ -165,6 +178,24 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 				Help: "The number of times to replicate incoming write requests.",
 			},
 		),
+		configuredTenantLimit: promauto.With(registerer).NewGauge(
```

Comment:
I think it's time to perhaps move it to it's own struct, especially given races in map we have to stop and background querying.

Reply:
Have moved all the limiting logic to a separate struct in https://github.com/thanos-io/thanos/pull/5520/commits/a8449564cda0fd8e139a9e7443090dfcdb443fdb 🙂